### ISSUE TITLE
Column filter "startsWith" defaults to "startsWithUTF8"

### ISSUE THREAD

**Issue by djuarezg** [ClickHouse/metabase-clickhouse-driver#224]
### Steps to reproduce
1. Set up a CH database
2. Go to as a question on a CH table
3. Click on filter for a string column
4. Apply a "Starts with" filter on the given column
5. 
![image](https://github.com/ClickHouse/metabase-clickhouse-driver/assets/34030545/2dac4713-2abf-43f1-9444-808487193d07)

If you go to the generated SQL question and edit the used function yourself all works fine

### Expected behaviour

Do not use `startsWithUTF8` on the applied filter when using filters through the UI instead of writing the full SQL question, or at least do not take for granted that CH has that function which is only available from https://clickhouse.com/docs/en/whats-new/changelog/2023#238 version

### Error log

```
Code: 46. DB::Exception: Unknown function startsWithUTF8. Maybe you meant: ['startsWith']: While processing startsWithUTF8(lowerUTF8(deviceTypes), lowerUTF8('desktop')). (UNKNOWN_FUNCTION) (version 23.3.2.37 (official build))\n, server ClickHouseNode [uri=https://bimbadabum:9001/default, options={sslrootcert=/etc/certs/cert_ca.crt,socket_timeout=42000000,connect_timeout=42000000,receive_timeout=42000000,send_timeout=42000000,use_server_time_zone_for_dates=true,use_no_proxy=true,product_name=metabase/1.3.3}]@-1310317476
```

#### Environment
* metabase-clickhouse-driver version: 1.3.3
* Metabase version: v0.48.3
* OS: Metabase Docker image

#### ClickHouse server
* ClickHouse Server version: 23.3.2.37
* `LowCardinality(Nullable(String))` fields

**Comment by slvrtrn** [comment]
What would be the preferred solution, given that [23.3 should be EOL when 24.3 is out](https://clickhouse.com/docs/en/faq/operations/production#how-to-choose-between-clickhouse-releases) and it's better to keep the Unicode versions of the `startWith`/`endWith`/`lower`/`position` functions in the driver by default? 

Maybe some kind of a driver configuration toggle (indicating that the ClickHouse deployment is <23.8)?

**Comment by djuarezg** [comment]
if the driver can identify somehow the version that would be great, otherwise a user-defined flag or conf would be sufficient.

**Comment by slvrtrn** [comment]
> if the driver can identify somehow the version that would be great

Yes, looks like it is the best solution. I'll check if this is possible to do from the driver in a clean way.

**Comment by slvrtrn** [comment]
Should be fixed in [1.3.4](https://github.com/ClickHouse/metabase-clickhouse-driver/releases/tag/1.3.4).
